velox/functions/prestosql/aggregates/ArrayAggAggregate.cpp

```diff
@@ -267,4 +267,35 @@ void registerArrayAggAggregate(
       overwrite);
 }
 
+void registerInternalArrayAggAggregate(
+    const std::string& prefix,
+    bool withCompanionFunctions,
+    bool overwrite) {
+  std::vector<std::shared_ptr<exec::AggregateFunctionSignature>> signatures{
+      exec::AggregateFunctionSignatureBuilder()
+          .typeVariable("E")
+          .returnType("array(E)")
+          .intermediateType("array(E)")
+          .argumentType("E")
+          .build()};
+
+  auto name = prefix + "$internal$array_agg";
```

Comment:
This is intended. We need an array_agg function that never ignores NULL. The built-in array_agg function cannot be used because it could ignore NULLs if the corresponding query config is set.